### PR TITLE
Remove legacy CEED_BASIS_COLLOCATED

### LINKED ISSUE #1355: Remove CEED_BASIS_COLLOCATED
We should remove `CEED_BASIS_COLLOCATED` for the v0.13 release in the spring

### PR DESCRIPTION
**Pull request by jeremylt**
Closes #1355

Drop legacy `CEED_BASIS_COLLOCATED`

MFEM now uses `CEED_BASIS_NONE`:
https://github.com/mfem/mfem/pull/3854

PETSc now uses `CEED_BASIS_NONE`:
https://gitlab.com/petsc/petsc/-/merge_requests/7125

### CODE REVIEW

**Comment by jeremylt** [comment]
PETSc PR merged, this is now good to merge

**Review by jrwrigh — APPROVED** [review]

**Comment by jeremylt** [comment]
shame on me - I forgot to update the release notes

Will merge once CI passes.

**Comment by jrwrigh** [comment]
Probably should've caught that myself. RIP.

**Comment by jeremylt** [comment]
Friday brains

File: include/ceed/ceed.h [hunk 1]
```diff
@@ -220,8 +220,6 @@ CEED_EXTERN const CeedVector CEED_VECTOR_NONE;
 /// @ingroup CeedBasis
 CEED_EXTERN const CeedBasis CEED_BASIS_NONE;
 
-CEED_EXTERN const CeedBasis CEED_BASIS_COLLOCATED;
-
```
Comment (jedbrown) [inline include/ceed/ceed.h:224]: It's late now, but I think we could have deprecated this initially. https://gcc.gnu.org/onlinedocs/gcc/Common-Variable-Attributes.html#index-deprecated-variable-attribute

Mohadese just encountered use of the old name in RDycore, which has been building without warnings for those using release libCEED via PETSc (even though `CEED_BASIS_COLLOCATED` was documented as deprecated by then).

**Pull request merged**